### PR TITLE
Remove DataType::Utf8

### LINKED ISSUE #1603: [Refactors types] Remove DataType::Utf8
**Summary**

Tasks to be done:

- [ ] Remove DataType::Utf8
- [ ] Rename `DataType::Binary` to be `DataType::String`, `DataType::String` is the logic datatype of `DataType::LargeBinary` in arrow.
- [ ] Refactor other functions to work with `DataType::String`

### PR DESCRIPTION
**Pull request by zhyass**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

- [x] https://github.com/jorgecarleitao/arrow2/issues/345
- [x] Remove DataType::Utf8
- [x] Rename `DataType::Binary` to be `DataType::String`, `DataType::String` is the logic datatype of `DataType::LargeBinary` in arrow.
- [x] Refactor other functions to work with `DataType::String`

## Changelog

- New Feature

## Related Issues

Fixes #1603

### CODE REVIEW

File: query/src/sessions/settings.rs [hunk 1]
```diff
@@ -27,11 +27,11 @@ pub struct Settings {
 
 impl Settings {
     apply_macros! { apply_getter_setter_settings, apply_initial_settings, apply_update_settings,
-        ("max_block_size", u64, 10000, "Maximum block size for reading".to_string()),
-        ("max_threads", u64, 16, "The maximum number of threads to execute the request. By default, it is determined automatically.".to_string()),
-        ("flight_client_timeout", u64, 60, "Max duration the flight client request is allowed to take in seconds. By default, it is 60 seconds".to_string()),
-        ("min_distributed_rows", u64, 100000000, "Minimum distributed read rows. In cluster mode, when read rows exceeds this value, the local table converted to distributed query.".to_string()),
-        ("min_distributed_bytes", u64, 500 * 1024 * 1024, "Minimum distributed read bytes. In cluster mode, when read bytes exceeds this value, the local table converted to distributed query.".to_string())
+        ("max_block_size", u64, 10000, "Maximum block size for reading".as_bytes().to_vec()),
```
Comment (sundy-li) [inline query/src/sessions/settings.rs:30]: just keep it as `&str`.

we can modify inside the `pub fn try_set_xx` functions.

Comment (zhyass) [reply query/src/sessions/settings.rs]: fixed

**Comment by zhyass** [comment]
Wait the pr https://github.com/jorgecarleitao/arrow2/pull/346

**Comment by sundy-li** [comment]
> Wait the pr jorgecarleitao/arrow2#346 complete

No need to wait, we can review and merge in previous and update the commit  in another pr.

**Comment by zhyass** [comment]
> No need to wait, we can review and merge in previous and update the commit in another pr.

Sure, opened

**Review by sundy-li — APPROVED** [review]
LGTM

**Pull request merged**